assets/js/components/OverlayCard/index.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * OverlayCard component.
+ *
+ * Site Kit by Google, Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { Slide } from '@material-ui/core';
+import classnames from 'classnames';
+import PropTypes from 'prop-types';
+
+/**
+ * Internal dependencies
+ */
+import { BREAKPOINT_SMALL, useBreakpoint } from '../../hooks/useBreakpoint';
+import MainBody from './MainBody';
+
+export default function OverlayCard( {
+	className,
+	title,
+	description,
+	ctaButton,
+	dismissButton,
+	GraphicDesktop,
+	GraphicMobile,
+	visible,
+} ) {
```

Comment:
Let's define the argument as `props` instead of destructing it. This will let us pick component properties and spread it in the `<MainBody ...>` call like this:

```js
const { visible, ...cardProps } = props;

...

return <MainBody ...cardProps />;
```